### PR TITLE
add patch for pybind11 2.6.0 to fix failing test due to extra whitespace

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)

cfr. https://github.com/pybind/pybind11/issues/3680

### CODE REVIEW

File: easybuild/easyconfigs/p/pybind11/pybind11-2.6.0-GCCcore-10.2.0.eb [hunk 1]
```diff
@@ -9,7 +13,12 @@ toolchain = {'name': 'GCCcore', 'version': '10.2.0'}
 
 source_urls = ['https://github.com/pybind/pybind11/archive/']
 sources = ['v%(version)s.tar.gz']
-checksums = ['90b705137b69ee3b5fc655eaca66d0dc9862ea1759226f7ccd3098425ae69571']
+patches = ['1d3b04e805c8a7dd4d8b68e2a311c75a66cae1d5.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/p/pybind11/pybind11-2.6.0-GCCcore-10.2.0.eb:16]: @sassy-crick Please rename the patch file to something like `pybind11-2.6.0_fix-test-extra-whitespace.patch`

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen2

**Review by smoors — APPROVED** [review]
lgtm

**Review by smoors — APPROVED** [review]

**Comment by smoors** [comment]
Going in, thanks @sassy-crick!

**Pull request merged**